templates/hooks/db_instance/delta_pre_compare.go.tpl

```diff
@@ -0,0 +1,12 @@
+    // Do not consider AvailabilityZone field for delta if it is missing in
+    // desired(a) and is present in latest(b) because AvailabilityZone field is
+    // late-initialized
+    // This special handling is only needed for DBInstance because a requeue
+    // error needs to be returned even after successful ModifyDBInstance call.
+    // See sdk_update_post_set_output.go.tpl for more details.
+    // The requeue error from update prevents the late initialization in
+    // reconciler.go and without ignoring AvailabilityZone for delta , the
+    // reconciler will keep updating the dbinstance and constantly requeueing it.
+    if a != nil && a.ko.Spec.AvailabilityZone == nil && b != nil && b.ko.Spec.AvailabilityZone != nil {
+        a.ko.Spec.AvailabilityZone = b.ko.Spec.AvailabilityZone
```

Comment:
isn't this what lateInitialize is for?

Reply:
Yep, but when the requeue error is returned from the update method lateInitialization is skipped.

If the above hook is not present then following steps occur:
Update -> Requeue -> Delta found for AZ field -> Update -> Requeue -> Delta found for AZ

With the above hook the following steps occur:
Update -> Requeue -> No Delta found for AZ field -> No Update -> LateInitialize -> Synced False -> Requeue -> No Delta -> LateInitialize ...

Reply:
I see... seems like a better long-term solution would be to have `lateInitialize` execute when a requeue error is returned from `Update`, no?

Reply:
Yes. Definitely. Requeue errors can be handled differently than other errors for create/update. I did not wish to include runtime change as part of this PR so did not go down that route.